osu.Game/Beatmaps/Drawables/DifficultyIconTooltip.cs

```diff
@@ -146,7 +146,10 @@ public void SetContent(DifficultyIconTooltipContent content)
             approachRate.Text = @" AR: " + adjustedDifficulty.ApproachRate.ToString(@"0.##");
             overallDifficulty.Text = @" OD: " + adjustedDifficulty.OverallDifficulty.ToString(@"0.##");
 
-            length.Text = "Length: " + TimeSpan.FromMilliseconds(displayedContent.BeatmapInfo.Length / rate).ToString(@"mm\:ss");
+            TimeSpan lengthTimeSpan = TimeSpan.FromMilliseconds(displayedContent.BeatmapInfo.Length / rate);
+            length.Text = "Length: " + (lengthTimeSpan.Hours > 0
```

Comment:
something like this would be cleaner

```cs
length.Text = "Length: " + lengthTimeSpan.ToString(lengthTimeSpan.TotalHours >= 1 ? @"hh\:mm\:ss" : @"mm\:ss");
```

Reply:
makes sense